core/templates/pages/exploration-player-page/services/exploration-recommendations.service.ts

```diff
@@ -79,37 +56,23 @@ export class ExplorationRecommendationsService {
       authorRecommendedExpIds: string[],
       includeAutogeneratedRecommendations: boolean,
       successCallback: (value?: Object | PromiseLike<Object>) => void): void {
-    let recommendationsUrlParams: RecommendationsUrlParams = {
-      stringified_author_recommended_ids: JSON.stringify(
-        authorRecommendedExpIds),
-      include_system_recommendations: 'false'
-    };
-
     let collectionId = this.urlService.getCollectionIdFromExplorationUrl();
     let storyId = this.urlService.getUrlParams().story_id;
     let currentNodeId = this.urlService.getUrlParams().node_id;
-    if (collectionId !== null) {
-      recommendationsUrlParams.collection_id = collectionId;
-    }
-    if (storyId !== null) {
-      recommendationsUrlParams.story_id = storyId;
-    }
-    if (currentNodeId !== null) {
-      recommendationsUrlParams.current_node_id = currentNodeId;
-    }
+    let explorationId = ExplorationRecommendationsService.explorationId;
+
+    let includeSystemRecommendations = 'false';
 
     if (
       includeAutogeneratedRecommendations &&
       !ExplorationRecommendationsService.isInEditorPage) {
-      recommendationsUrlParams.include_system_recommendations = 'true';
+      includeSystemRecommendations = 'true';
     }
 
-    this.http.get(
-      '/explorehandler/recommendations/' +
-      ExplorationRecommendationsService.explorationId, {
-        params: recommendationsUrlParams
-      }).toPromise().then((data: IExplorationSummaries) => {
-      successCallback(data.summaries);
+    this.expRecommendationBackendApiService.getRecommendedSummaryDicts(
+      authorRecommendedExpIds, includeSystemRecommendations, collectionId,
+      storyId, currentNodeId, explorationId).then((expSummaries) => {
```

Comment:
Can we just do `expSummaries => ..` instead?

Reply:
done